.travis.yml

```diff
@@ -1,106 +1,38 @@
-sudo: false
+# We will use docker to set up our environment, so don't use any particular
+# language in Travis itself
+language: generic
 
-language: cpp
+# Enable docker
+sudo: required
+services:
+    - docker
 
-os:
-  # - linux
-  - osx
+# Disable automatic submodule fetching ...
+git:
+    submodules: false
 
-osx_image: xcode8
+# ... in order to ensure we do a non-shallow submodule fetch
+before_install:
+    - git fetch --unshallow --tags
```

Comment:
The "nasty issues" were that `git describe` would fail.  IIRC the `--unshallow` was to make sure that we could do the `git describe HEAD^` call used when generating the changelog, while the `--tags` was to make sure we had all upstream tags (which might not be such a big deal, but had some irritations).